inc/zoo/swar/SWAR.h

```diff
@@ -73,10 +73,14 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+          constexpr auto NBitsLessThanByteOfT = NBits < sizeof(T) * 8;
+          if constexpr (NBitsLessThanByteOfT) {
```

Comment:
Why using `if constexpr`  while you could use the ternary operator, `?`?

Reply:
this apparently was required to appease the compiler.